Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/StatusBean.java

```diff
@@ -180,4 +203,37 @@ private void initPieChart() {
         refreshPieChart();
     }
 
+    public List<CertificateDTO> getCertificateInfo() {
+        try {
+            certificateList = new ArrayList<>();
+            addCertificates(certificateService.fetchKeyStores(), certificateList);
+            addCertificates(certificateService.fetchTrustStores(), certificateList);
+
+            Collections.sort(certificateList, new Comparator<CertificateDTO>() {
+                @Override
+                public int compare(CertificateDTO o1, CertificateDTO o2) {
+                    return o2.getExpiresInDays() > o1.getExpiresInDays() ? -1
```

Comment:
I see what you're doing here, but it might be simpler ( and *much* easier to read ) to subtract the number of days from one another. Since Comparator doesn't need to explicitly have -1, 0, or 1, we can subtract the days which will give us a positive or negative number, or 0 if they are equivalent. Thus all 3 cases are handled.

Reply:
why do you want to subtract and get the number of days since the operator( > or = ) can get the result?

Reply:
Having a ? operator nested inside of another ? operator is difficult to read in my opinion, and it appears to be only for the sake of explicitly returning -1, 0, or 1, which isn't necessary

Subtracting the days remaining is much simpler to read and will satisfy the requirements.
For example, if one object evaluated to 15, and another evaluated to 10, we would have the following situations:
`
Less than: 10 - 15 = -5
Greater than: 15 - 10 = 5
If we had 2 objects with similar dates: 10-10 = 0
`

Since the comparator only uses the sign of the integer for sorting, we don't have to convert the 5 and -5 to 1 and -1 respectively, and all 3 cases are covered with respect to the required contract for comparators leaving us with a very nice and simple implementation:

`return o1.getExpiresInDays() - o2.getExpiresInDays()`

Reply:
ok sure, addressing this.